### PR TITLE
[GCE] Support legacy-https and generic health checks

### PR DESCRIPTION
**Pull request by nicksardo**
**What this PR does / why we need it**:
- Adds CRUD functions to manage `compute.HttpsHealthChecks` 
The legacy HTTPS healthchecks will be used by the GLBC (GCE Load balancer Controller)

- Adds CRUD functions to manage `compute.HealthChecks`
These are required for the internal load balancer

- Removes the logic that disregards NotFound errors on DeleteHttpHealthChecks as this is useful information for callers. Here are the three known invocations within kubernetes: 
[gce/gce_loadbalancer.go#L457](https://github.com/kubernetes/kubernetes/blob/bc6e77d42fdb40bb82796fd8ee8ba2b04dd9a0ea/pkg/cloudprovider/providers/gce/gce_loadbalancer.go#L457): Only prints warning that HC wasn't deleted  -> acceptable
[gce/gce_loadbalancer.go#L465](https://github.com/kubernetes/kubernetes/blob/bc6e77d42fdb40bb82796fd8ee8ba2b04dd9a0ea/pkg/cloudprovider/providers/gce/gce_loadbalancer.go#L465): Err is ignored if not nil  -> acceptable
[e2e/framework/ingress_utils.go#L530](https://github.com/kubernetes/kubernetes/blob/bc6e77d42fdb40bb82796fd8ee8ba2b04dd9a0ea/test/e2e/framework/ingress_utils.go#L530): Already checks if is NotFound error -> acceptable

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*:
Step one of https://github.com/kubernetes/ingress/issues/494
Step one of #33483 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nicksardo** [comment]
fyi @bowei

**Comment by nicksardo** [comment]
@k8s-bot cvm gce e2e test this

File: pkg/cloudprovider/providers/gce/gce_healthchecks.go [hunk 1]
```diff
@@ -59,8 +52,88 @@ func (gce *GCECloud) CreateHttpHealthCheck(hc *compute.HttpHealthCheck) error {
...
+// ListHttpHealthChecks lists all HttpHealthChecks in the project.
 func (gce *GCECloud) ListHttpHealthChecks() (*compute.HttpHealthCheckList, error) {
 	// TODO: use PageToken to list all not just the first 500
 	return gce.service.HttpHealthChecks.List(gce.projectID).Do()
 }
+
+// Legacy HTTPS Health Checks
+
+// GetHttpsHealthCheck returns the given HttpsHealthCheck by name.
+func (gce *GCECloud) GetHttpsHealthCheck(name string) (*compute.HttpsHealthCheck, error) {
```
Comment (bowei) [inline pkg/cloudprovider/providers/gce/gce_healthchecks.go]: Should we put Legacy in the name? Might be confusing unless the user reads the comment above

Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/gce_healthchecks.go]: Hmm, only the GCP documentation refers to these health checks as legacy. The network load balancer still requires HttpHealthCheck for probing. I'm not sure if adding "Legacy" is necessary at this point (but will be some day). We could add "(Legacy)" in each func comment.

Comment (bowei) [inline pkg/cloudprovider/providers/gce/gce_healthchecks.go]: ok, SG

**Pull request merged**